CRM/Contact/Form/Edit/TagsAndGroups.php

```diff
@@ -137,6 +137,9 @@ public static function buildQuickForm(
 
       if (!empty($tags)) {
         $form->add('select2', 'tag', ts('Tag(s)'), $tags, FALSE, ['class' => 'huge', 'placeholder' => ts('- select -'), 'multiple' => TRUE]);
+        if ($isRequired) {
+          $form->addRule('tag', ts('Tag is a required field.'), 'required');
```

Comment:
Can you just change param 5 from FALSE to $isRequired? I haven't tested but it seems like that would do the same thing as the `if` and addRule.

For unit testing you could try using `$form = $this->getFormObject('CRM_Profile_Form_Edit', [params])`. I don't know offhand if there's an existing test that is similar - sometimes you have to fiddle a bit with the params and `$_GET` to make it work for the given form, but then you can do stuff like call $form->buildForm(), $form->postProcess(), etc.